app/src/main/java/tech/ula/utils/ServerUtility.kt

```diff
@@ -57,13 +55,19 @@ class ServerUtility(
     }
 
     /**
-     * Execute the boot script, should be called only when the session has the startOnBoot flag
-     * Users can use this to add their own custom commands to this script.
+     * Execute the user-defined startCommand
+     * Creates a new script based on userland_profile.sh and adds the user-defined command to the script
      */
-    private fun runBootScript(session: Session) {
+    fun executeStartCommand(session: Session) {
+        if (session.startCommand.isEmpty())
+            return
         val filesystemDirName = session.filesystemId.toString()
-        // TODO create new session parameter and turn this into a user-defined command
         val command = "/support/autostart.sh"
+        val startScript = File(applicationFilesDirPath, "$filesystemDirName$command")
+        val profileScript = File(applicationFilesDirPath, "$filesystemDirName/support/userland_profile.sh")
+        profileScript.copyTo(startScript, overwrite = true)
+        startScript.appendText("\n${session.startCommand}")
+        startScript.setExecutable(true, false)
```

Comment:
I'd like to see named parameters for the boolean values in `setExecutable` like we talked about previously. Just helps people reading the code understand the call without needing to look up documentation.

Reply:
Not possible in this case. 
Only possible when calling Kotlin functions and the File class is Java.

Reply:
It could be useful to define the booleans as variables before the call then. 

```
val isExecutable = true
val isOwnerOnly = false
startScript.setExecutable(isExecutable, isOwnerOnly)
```

That's pretty nitpicky though, so not a required change.